src/react/elements.js

```diff
@@ -164,3 +173,66 @@ export function createInternalReactElement(
   obj.makeFinal();
   return obj;
 }
+
+type ElementTraversalVisitor = {
+  visitType: (typeValue: Value) => void,
+  visitKey: (keyValue: Value) => void,
+  visitRef: (keyValue: Value) => void,
+  visitAbstractOrPartialProps: (propsValue: AbstractValue | ObjectValue) => void,
+  visitConcreteProps: (propsValue: ObjectValue) => void,
+  visitChildNode: (childValue: Value) => void,
+};
+
+export function traverseReactElement(
+  realm: Realm,
+  reactElement: ObjectValue,
+  traversalVisitor: ElementTraversalVisitor
```

Comment:
I considered putting these methods on the React serializer/visitor itself but it felt way too abstract and indirect, especially because React visitor is already a "visitor". So I wanted to make the relationship very direct by making this an argument with a special type.